### ISSUE TITLE
RustRover 2024.2 EAP Compatibility

### ISSUE THREAD

**Issue by farces** [FalsePattern/ZigBrains#60]
Hello, 

The plugin version/compatibility matrix lists RustRover 2024.2 as supported, however enabling the plugin prevents the current project from being closed (if one was open), and when creating and opening a new project the IDE throws an error before it's able to read the project structure. Switching ZigBrains off and restarting the IDE lets the project load, but re-enabling ZigBrains causes the same symptoms to reoccur.

Looking at the files created the only thing that looks odd is that the toolchain home directory for Zig is relative to the project path instead of being absolute (it is absolute under languages configuration in RR), but I don't think that's related:
`<option name="toolchainHomeDirectory" value="$PROJECT_DIR$/../../zig/zig-windows-x86_64-0.14.0-dev.1057+76f062690" />`

The Zig project created by RustRover doesn't look like it has had `zig init` run on it either, but doing that manually doesn't resolve the issue unfortunately.

The errors point to an internal IDE function getSourceRoots() failing:
```
[79681fe1] Failed to execute task com.intellij.util.indexing.UnindexedFilesScannerExecutorImpl$ScheduledScanningTask@40261211

java.lang.RuntimeException: java.util.concurrent.ExecutionException: java.lang.NullPointerException: getSourceRoots(...) must not be null
```

**OS:** Windows 11
**Zig:** 0.14.0-dev.1057
**ZLS:** 0.14.0-dev

**Comment by farces** [comment]
Checked a few other issues on this project and found this note about relative paths:
              Looks like your standard library location is a relative path, it needs to be an absolute path
![image](https://github.com/user-attachments/assets/d55f9627-191a-49af-8fef-7ee176811ff3)

_Originally posted by @FalsePattern in https://github.com/FalsePattern/ZigBrains/issues/56#issuecomment-2251223985_

I went and changed the stdlib path from the default to override and manually selected the stdlib path and everything works now. Seems like the default path selection might not be working correctly, maybe not in all cases but for mine at least:
My project structure for reference is:
**Project:** D:\code\RustProjects\zigtest
**Zig root:** D:\code\zig\zig-windows-x86_64-0.14.0-dev.1057+76f06269

And as mentioned in the issue it looks like `toolchainHomeDirectory` in zigbrains.xml is relative to the project path.

**Comment by FalsePattern** [comment]
Should be fixed in 17.1.0

**Comment by farces** [comment]
Have tested 17.1.0 and can confirm it works however I had to delete my .idea directory, reload RustRover and reconfigure paths for the plugin; just updating actually broke my working config from 17.0.0 (with override stdlib path to be absolute), and redoing paths under Languages->Zig wasn't resolving it.

Thanks for the fix regardless fp!

**Comment by FalsePattern** [comment]
Should be properly fixed in 17.2.0.
